app/assets/javascripts/discourse/tests/integration/components/file-size-input-test.js

```diff
@@ -0,0 +1,84 @@
+import { click, fillIn, render } from "@ember/test-helpers";
+import { hbs } from "ember-cli-htmlbars";
+import { module, test } from "qunit";
+import { setupRenderingTest } from "discourse/tests/helpers/component-test";
+import selectKit from "discourse/tests/helpers/select-kit-helper";
+
+module("Integration | Component | file-size-input", function (hooks) {
+  setupRenderingTest(hooks);
+
+  test("file size unit selector", async function (assert) {
+    this.set("value", 4096);
+    this.set("max", 8192);
+    this.set("onChangeSize", function () {});
+    this.set("updateValidationMessage", function () {});
+
+    await render(hbs`
+      <FileSizeInput
+        @sizeValueKB={{readonly this.value}}
+        @classNames="file-size-input-test"
+        @onChangeSize={{this.onChangeSize}}
+        @updateValidationMessage={{this.updateValidationMessage}}
+        @max=4096
+        @message=""
+      />
+    `);
+
+    await click(".file-size-unit-selector");
+
+    await selectKit(".file-size-unit-selector").expand();
+    await selectKit(".file-size-unit-selector").selectRowByValue("kb");
+
+    let fileSize = this.element.querySelector(".file-size-input").value;
+
+    assert.equal(fileSize, 4096, "value is changed when the unit is changed");
```

Comment:
you can use qunit-dom to assert that too:

```suggestion
    assert
      .dom(".file-size-input")
      .hasValue(4096, "value is changed when the unit is changed");
```